MetaMask/metamask-mobile
# chore: Implements push notifications FCM support

## **Description**

This PR implements support to FCM (Firebase Cloud Messaging) on both iOS and Android on mobile. 

## **Related issues**

Fixes: [NOTIFY-1219](https://consensyssoftware.atlassian.net/browse/NOTIFY-1219?atlOrigin=eyJpIjoiZDJiMzczYTc5ZGI5NGU3OWE4YTNjZDAyN2M0ZjIzYWYiLCJwIjoiaiJ9)

## **Manual testing steps**

1. Go to Settings --> Notifications --> Enable
2. Open a separated device (or extension) repeat the process above
3. Do any transaction between the two clients/devices
4. Leave the transaction recipient open to see it in Foreground
5. Put app in background to see a native notification

## **Screenshots/Recordings**



### **Before**



### **After**



https://github.com/user-attachments/assets/58f7adea-076e-4703-ab0b-58d78144b524


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**



[NOTIFY-1219]: https://consensyssoftware.atlassian.net/browse/NOTIFY-1219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
